doc/develop-plugin.md

```diff
@@ -0,0 +1,159 @@
+
+### Extending existing plugins vs. Creating new plugins
+
+Adding a new platform support to an existing plugin is simple: create a platform package that implements the platform interface of the plugin. It is technically not required for the original plugin author to _endorse_ your new platform implementation. A developer can use the unendorsed platform implementation in their app, but must specify both package dependencies in the app's pubspec file. For example, if there is a `foobar_tizen` implementation for the `foobar` plugin, the app's pubspec file must include both the `foobar` and `foobar_tizen` dependencies, unless the original `foobar` author adds `foobar_tizen` as a dependency in the pubspec of `foobar`.
+
+Note: Even if the original plugin is not a federated plugin (has no platform interface package), it is possible to create an unendorsed platform implementation of the plugin by implicitly implementing the plugin's platform channels.
+
+You can create a new plugin from scratch, if the functionality you want to implement is not implemented by any other plugin, or is specific to Tizen. The new plugin can be either a single package plugin or a federated plugin, depending on whether you want to target the Tizen platform only or other platforms as well.
+
+## Create a plugin package
+
+If you're to extend an existing plugin for Tizen, it is common to add the `_tizen` suffix to the package name:
+
+```sh
+flutter-tizen create --template plugin foobar_tizen
+```
+
+Otherwise, follow the `lowercase_with_underscores` style convention to name your package:
+
+```sh
+flutter-tizen create --template plugin plugin_name
+```
+
+Once the package is created, you will be prompted to add some information to its pubspec file. Open the main `plugin_name/` directory in VS Code, locate the `pubspec.yaml` file, and replace the `some_platform:` map with `tizen:` as suggested by the tool. This information is needed by the flutter-tizen tool to find and register the plugin when building an app that depends on the plugin.
+
+```yaml
+The `pubspec.yaml` under the project directory must be updated to support Tizen.
+Add below lines to under the `platforms:` key.
+
+tizen:
+  pluginClass: PluginNamePlugin
+  fileName: plugin_name_plugin.h
+```
+
+The created package contains an example app in the `example/` directory. You can run it using the `flutter-tizen run` command:
+
+```sh
+$ cd plugin_name/example
+$ flutter-tizen run
+```
+
+## Implement the plugin
+
+### 1. Define the package API (`.dart`)
+
+The API of the plugin package is defined in Dart code. Locate the file `lib/plugin_name.dart` in VS Code, and then you will see the `platformVersion` method defined in the plugin main class. Invoking this method will invoke the `getPlatformVersion` method through a method channel named `plugin_name`. This file is not required if you're extending an existing plugin for Tizen, so you can safely remove it.
```

Comment:
It would be nice if the description that the `platformVersion` getter of the .dart file is just an example is added.

Reply:
I made some change. Thanks!